File: darwin/flutter_compress_pro/Sources/flutter_compress_pro/ErrorCode.swift
/// Channel error codes. Mirrors `lib/src/error_codes.dart` and `ErrorCode.kt` —
/// these strings are the public contract, so keep all three in sync and treat a
/// change to any value as a breaking change.
enum ErrorCode {
  static let cancelled = "cancelled"

  // Video.
  static let infoFailed = "info_failed"
  static let estimateFailed = "estimate_failed"
  static let compressFailed = "compress_failed"
  static let thumbnailFailed = "thumbnail_failed"
  static let saveFailed = "save_failed"

  // Images.
  static let imageInfoFailed = "image_info_failed"
  static let imageCompressFailed = "image_compress_failed"

  // Plumbing.
  static let badArguments = "bad_arguments"
  static let noEngine = "no_engine"
  static let unsupported = "unsupported"
  /// Android-only today; declared here so all three code sets match.
  static let permissionDenied = "permission_denied"
}

File: android/src/main/kotlin/com/compress/all/flutter_compress_pro/ErrorCode.kt
package com.compress.all.flutter_compress

/**
 * Channel error codes. Mirrors `lib/src/error_codes.dart` and `ErrorCode.swift` —
 * these strings are the public contract, so keep all three in sync and treat a
 * change to any value as a breaking change.
 */
internal object ErrorCode {
    const val CANCELLED = "cancelled"

    // Video.
    const val INFO_FAILED = "info_failed"
    const val ESTIMATE_FAILED = "estimate_failed"
    const val COMPRESS_FAILED = "compress_failed"
    const val THUMBNAIL_FAILED = "thumbnail_failed"
    const val SAVE_FAILED = "save_failed"

    // Images.
    const val IMAGE_INFO_FAILED = "image_info_failed"
    const val IMAGE_COMPRESS_FAILED = "image_compress_failed"

    // Plumbing.
    const val NO_ENGINE = "no_engine"
    const val BAD_ARGUMENTS = "bad_arguments"
    const val UNSUPPORTED = "unsupported"
    const val PERMISSION_DENIED = "permission_denied"
}

/** A required channel argument was absent or had the wrong type. */
internal class BadArgumentException(key: String) :
    IllegalArgumentException("Missing or invalid argument: $key")
